### PR TITLE
[docs] hba fixes

### PR DESCRIPTION
**Pull request by ddhodge**
Edits to host based authentication (#9095)

### CODE REVIEW

**Review by ssung-yugabyte — APPROVED** [review]
Yb also support "local" (unix socket) in hba.  eg 
local      database  user  auth-method  [auth-options]

we can add this to replace
host      database   user CIDR auth-method [auth-options]

File: docs/content/latest/secure/authentication/host-based-authentication.md [hunk 1]
```diff
@@ -25,54 +25,70 @@ showAsideToc: true
...
 # This is an autogenerated file, do not edit manually!
-host all all 0.0.0.0/0 trust
-host all all ::0/0 trust
+host all yugabyte 127.0.0.1/0 trust
+host all yugabyte ::1/128 trust
 /Users/yugabyte/yugabyte-data/node-1/disk-1/pg_data/ysql_hba.conf (END)
 ```
 
-## Record formats
+Because `ysql_hba.conf` records are examined sequentially for each connection attempt, the order of the records is significant. Typically, earlier records will have tight connection match parameters and weaker authentication methods, while later records will have looser match parameters and stronger authentication methods. For example, you might want to use trust authentication for local TCP/IP connections, but require a password for remote TCP/IP connections. In this case, a record specifying `trust` authentication for connections from `127.0.0.1` would appear before a record specifying password authentication for a wider range of allowed client IP addresses.
```
Comment (lizayugabyte) [inline docs/content/latest/secure/authentication/host-based-authentication.md:84]: Typically, earlier records will have tight connection match parameters and weaker authentication methods, while later records will have looser match parameters and stronger authentication methods. --> Typically, earlier records have tight connection match parameters and weaker authentication methods, while later records have looser match parameters and stronger authentication methods.

File: docs/content/latest/secure/authentication/host-based-authentication.md [hunk 2]
```diff
@@ -81,160 +97,161 @@ hostssl    database  user  IP-address  netmask  auth-method  [auth-options]
 hostnossl  database  user  IP-address  netmask  auth-method  [auth-options]
 ```
 
-The meaning of these fields are described below.
+The fields are described below.
```
Comment (lizayugabyte) [inline docs/content/latest/secure/authentication/host-based-authentication.md:100]: Suggestion:  
rephrase to get rid of "below" (e.g. some flavour of "following")

File: docs/content/latest/secure/enable-authentication/ysql_hba_conf-configuration.md [hunk 3]
```diff
@@ -21,21 +21,25 @@ showAsideToc: true
   </li>
 </ul>
 
-The client authentication in YugabyteDB is managed by the T-Server <code>[--ysql_hba_conf_csv](../../../reference/configuration/yb-tserver/#ysql-hba-conf-csv) </code>
-configuration flag, which works similar to the <code>pg_hba.conf</code> file in PostgreSQL. 
+Client authentication in YugabyteDB is managed by the YB-TServer [`--ysql_hba_conf_csv`](../../../reference/configuration/yb-tserver/#ysql-hba-conf-csv) configuration flag, which works similar to the `pg_hba.conf` file in PostgreSQL.
```
Comment (lizayugabyte) [inline docs/content/latest/secure/enable-authentication/ysql_hba_conf-configuration.md:24]: which works similar to the `pg_hba.conf` file in PostgreSQL. --> which works similarly to the `pg_hba.conf` file in PostgreSQL.

**Review by lizayugabyte — COMMENTED** [review]
A few minor comments. 
Otherwise, it's good.

**Pull request merged**